static/app/api/apiOptions.ts

```diff
@@ -0,0 +1,66 @@
+import {queryOptions} from '@tanstack/react-query';
+
+import {fetchDataQuery, type QueryKeyEndpointOptions} from 'sentry/utils/queryClient';
+
+import type {ApiMapping, ApiPath, MaybeApiPath} from './apiDefinition';
+
+function safeEncodeURIComponent(input: string) {
+  try {
+    // If decoding and re-encoding gives the same string, it's already encoded
+    if (encodeURIComponent(decodeURIComponent(input)) === input) {
+      return input;
+    }
+  } catch (e) {
+    // decodeURIComponent threw an error, so it was not encoded
+  }
+  return encodeURIComponent(input);
+}
```

Comment:
not sure if this is really necessary, I just wanted to avoid duplicate encoding in case the params were already encoded on the call-sides.